cpp/include/cudf/datetime.hpp

```diff
@@ -37,6 +37,23 @@ namespace datetime {
  * @file
  */
 
+/**
+ * @brief Types of datetime components that may be extracted.
+ */
+enum class datetime_component {
```

Comment:
Give this a specific base - either `uint8_t` or `int32_t`.
```suggestion
enum class datetime_component : uint8_t {
```